censys/cli/commands/asm.py

```diff
@@ -0,0 +1,134 @@
+"""Censys ASM CLI."""
+import argparse
+import json
+import sys
+
+from rich.prompt import Prompt
+
+from censys.asm.seeds import SEED_TYPES, Seeds
+from censys.common.config import DEFAULT, get_config, write_config
+from censys.common.exceptions import CensysUnauthorizedException
+
+
+def cli_asm_config(_: argparse.Namespace):  # pragma: no cover
+    """Config asm subcommand.
+
+    Args:
+        _: Argparse Namespace.
+    """
+    api_key_prompt = "Censys ASM API Key"
+
+    config = get_config()
+    api_key = config.get(DEFAULT, "asm_api_key")
+
+    if api_key:
+        key_len = len(api_key) - 4
+        redacted_api_key = api_key.replace(api_key[:key_len], key_len * "*")
+        api_key_prompt = f"{api_key_prompt} [cyan]({redacted_api_key})[/cyan]"
+
+    api_key = Prompt.ask(api_key_prompt) or api_key
+
+    if not api_key:
+        print("Please enter valid credentials")
+        sys.exit(1)
+
+    try:
+        # Assumes that login was successfully
+        config.set(DEFAULT, "asm_api_key", api_key)
+
+        write_config(config)
+        print("\nSuccessfully configured credentials")
+        sys.exit(0)
+    except CensysUnauthorizedException:
+        print("Failed to authenticate")
+        sys.exit(1)
+
+
+def cli_add_seeds(args: argparse.Namespace):
+    """Add seed subcommand.
+
+    Args:
+        args (Namespace): Argparse Namespace.
+    """
+    if args.input_file:
+        if args.input_file == "-":
+            data = sys.stdin.read()
+        else:
+            with open(args.input_file, "r") as f:
+                data = f.read()
+    else:
+        data = args.json
+    try:
+        seeds = json.loads(data)
+    except json.decoder.JSONDecodeError as e:
+        print(f"Invalid json {e}")
+        sys.exit(1)
+
+    seeds_to_add = []
+    for seed in seeds:
+        if isinstance(seed, dict):
+            if "type" not in seed:
+                seed["type"] = args.deafult_type
+        elif isinstance(seed, str):
+            seed = {"value": seed, "type": args.deafult_type}
+        else:
+            print(f"Invalid seed {seed}")
+            sys.exit(1)
+        if "label" not in seed:
+            seed["label"] = args.label_all
+        seeds_to_add.append(seed)
+
+    s = Seeds(args.api_key)
+    res = s.add_seeds(seeds_to_add)
+    added_count = len(res["addedSeeds"])
+    added_str = None
+    if added_count < len(seeds_to_add):
+        added_str = f"{added_count}/{len(seeds_to_add)}"
```

Comment:
suggest something like "Added X of Y supplied seeds (seeds are not added if they already exist)"